readme.md

```diff
@@ -567,6 +547,58 @@ The numeric exit code of the subprocess that was run.
 
 This is `undefined` when the subprocess could not be spawned or was terminated by a [signal](#signal).
 
+#### pipedFrom
+
+Type: [`Array<SubprocessResult | ExecaError>`](#subprocessresult)
+
+Results of the other subprocesses that were [piped](#pipe-multiple-subprocesses) into this subprocess. This is useful to inspect a series of subprocesses piped with each other.
+
+This array is initially empty and is populated each time the [`.pipe()`](#pipefile-arguments-options) method resolves.
+
+### ExecaError
+### ExecaSyncError
+
+Type: `Error`
+
+Exception thrown when the subprocess [fails](#failed), either:
+- its [exit code](#exitcode) is not `0`
+- it was [terminated](#isterminated) with a [signal](#signal), including [`.kill()`](#killerror)
+- [timing out](#timedout)
+- [being canceled](#iscanceled)
+- there's not enough memory or there are already too many subprocesses
+
+This has the same shape as [successful results](#subprocessresult), with the following additional properties.
+
+#### message
+
+Type: `string`
+
+Error message when the subprocess failed to run. In addition to the [underlying error message](#originalMessage), it also contains some information related to why the subprocess errored.
+
+The subprocess [`stderr`](#stderr), [`stdout`](#stdout) and other [file descriptors' output](#stdio) are appended to the end, separated with newlines and not interleaved.
+
+#### shortMessage
+
+Type: `string`
+
+This is the same as the [`message` property](#message) except it does not include the subprocess [`stdout`](#stdout)/[`stderr`](#stderr)/[`stdio`](#stdio).
+
+#### originalMessage
+
+Type: `string | undefined`
+
+Original error message. This is the same as the `message` property excluding the subprocess [`stdout`](#stdout)/[`stderr`](#stderr)/[`stdio`](#stdio) and some additional information added by Execa.
+
+This is `undefined` unless the subprocess exited due to an `error` event or a timeout.
+
+#### cause
+
+Type: `unknown | undefined`
+
+Underlying error, if there is one. For example, this is set by [`.kill(error)`](#killerror).
+
+This is usually an `Error` instance.
```

Comment:
We could normalize `error.cause` to an `Error` instance. But then, users might miss some information. For example, if it is an object but not an `Error` instance, it would require some more complex logic to serialize its properties.

I have [a library](https://github.com/ehmicky/normalize-exception) that does this, and getting this correctly is actually tricky.

But since it is not guaranteed to be an `Error` instance, we document it so users do `error?.message` instead of something like `error.message`.

Reply:
I think it's fine to not guarantee it being an error. Even though it's suboptimal, it's how it works everywhere in JS.